pkg/storage/unified/resource/search.go

```diff
@@ -249,10 +266,32 @@ func (s *searchSupport) handleEvent(ctx context.Context, evt *WrittenEvent) {
 		return
 	}
 
-	err = index.Write(doc)
-	if err != nil {
-		s.log.Warn("error writing document watch event", "error", err)
-		return
+	switch evt.Type {
+	case WatchEvent_ADDED, WatchEvent_MODIFIED:
+		err = index.Write(doc)
+		if err != nil {
+			s.log.Warn("error writing document watch event", "error", err)
+			return
+		}
+		if evt.Type == WatchEvent_ADDED {
+			search.IndexMetrics.IndexedKinds.WithLabelValues(evt.Key.Resource).Inc()
+		}
+	case WatchEvent_DELETED:
```

Comment:
@ryantxu I don't think we were handling deleted events anywhere so I added that in here.